agent/fw_drupal8.c

```diff
@@ -605,10 +605,19 @@ NR_PHP_WRAPPER_END
 #endif  // OAPI
 
 static bool nr_is_invalid_key_val_arr(nr_php_string_hash_key_t* key,
-                                      zval* val) {
+                                      zval* val,
+                                      const char* key_ident) {
   if (NULL == key || 0 == ZEND_STRING_LEN(key)
       || 0 == nr_php_is_zval_valid_array(val)
       || 0 == zend_hash_num_elements(Z_ARRVAL_P(val))) {
+    if (NULL != key) {
```

Comment:
if you break up the if, you don't have to do the NULL check again.

could be something like:
```
if (NULL == key || 0 == ZEND_STRING_LEN(key)
invalid key msg
return false
if 0 == nr_php_is_zval_valid_array(val)
      || 0 == zend_hash_num_elements(Z_ARRVAL_P(val)))
invalid value msg 
return false

Reply:
726fb76db25e505ba800028cd4e04211c93b7369